otter/test/test_controller.py

```diff
@@ -1464,3 +1471,96 @@ def test_checks_pass_replace_true_no_purge_failure(self):
         ])
         self.assertRaises(ValueError, self._remove, True, False,
                           seq_dispatcher)
+
+    def test_perform_convergence_remove_from_group(self):
+        """
+        Perform :func:`convergence_remove_server_from_group` with the given
+        dispatcher.
+        """
+        self.state.desired = 2
+        fake_super = FakeSupervisor()
+
+        dispatcher = ComposedDispatcher([
+            test_dispatcher(),
+            SequenceDispatcher([
+                self._tenant_retry(
+                    get_server_details('server_id').intent,
+                    lambda _: (StubResponse(200, {}), self.server_details)),
+                (GetScalingGroupInfo(tenant_id='tenant_id',
+                                     group_id='group_id'),
+                    lambda _: (self.group, self.group_manifest_info)),
+                self._tenant_retry(
+                    mock.ANY,
+                    lambda i: sync_perform(dispatcher, Effect(i)))
+            ]),
+            get_eviction_dispatcher(fake_super)
```

Comment:
and I think that this dispatcher should be left out, and the result simulated in the performer on line 1494, instead of calling the real performer in the unit test.  This also allows us to get rid of the scrub_calls mocking in this test.

To explain the philosophy behind this: if the `EvictServerFromScalingGroup` intent is constructed correctly, and the performer for `EvictServerFromScalingGroup` has been separately tested, we should be confident that it will be performed correctly. We shouldn't need to cross this boundary in our test. That is pretty much the primary reason of having abstract, application-level intents like this. Otherwise we could just use `Func` everywhere and have our tests be just as side-effecty as they always were.